pkg/liqo-controller-manager/webhooks/shadowpod/peeringinfo_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2019-2022 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shadowpod
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	vkv1alpha1 "github.com/liqotech/liqo/apis/virtualkubelet/v1alpha1"
+)
+
+var _ = Describe("Peering Info", func() {
+	var (
+		spValidator                  *Validator
+		found                        bool
+		dryRun                       bool
+		cache                        *peeringCache
+		fakeNamespace                *corev1.Namespace
+		peeringInfo, peeringInfoTest *peeringInfo
+		spd                          *Description
+		err                          error
+		errTest                      error
+		fakeClient                   client.Client
+		shadowPod                    *vkv1alpha1.ShadowPod
+		containers                   []containerResource
+	)
+
+	BeforeEach(func() {
+
+		fakeNamespace = forgeNamespaceWithClusterID(clusterID)
+
+		fakeClient = fake.NewClientBuilder().WithScheme(scheme).WithObjects(fakeNamespace).Build()
+
+		spValidator = webhook.Admission{Handler: NewValidator(fakeClient, false)}.Handler.(*Validator)
+
+		cache = spValidator.PeeringCache
+
+		peeringInfoTest = createPeeringInfo(*clusterIdentity, *resourceQuota)
+
+		containers = nil
+
+		containers = append(containers, containerResource{cpu: int64(resourceCPU), memory: int64(resourceMemory)})
+
+		shadowPod = forgeShadowPodWithResourceLimits(containers, nil)
+
+		Expect(spValidator.InjectDecoder(decoder)).To(Succeed())
+	})
+
+	Describe("Get or Create a PeeringInfo", func() {
+		JustBeforeEach(func() {
+			peeringInfo = cache.getOrCreatePeeringInfo(*clusterIdentity, *resourceQuota)
+		})
+
+		When("The Peering Info exists", func() {
+			BeforeEach(func() {
+				peeringInfo = nil
+				cache.peeringInfo.Store(clusterID, createPeeringInfo(*clusterIdentity, *resourceQuota))
+			})
+			It("should return a peering info", func() {
+				Expect(peeringInfo).ToNot(BeNil())
+				Expect(peeringInfo).To(Equal(peeringInfoTest))
+			})
+		})
+		When("The Peering Info does not exist", func() {
+			BeforeEach(func() {
+				cache.peeringInfo.Delete(clusterID)
+				peeringInfo = nil
+			})
```

Comment:
This is not required, as already reset at the beginning of the test (same for the next test).